engine.go

```diff
@@ -267,3 +274,63 @@ func (e *EngineBackend) ForkchoiceUpdatedV1(ctx context.Context, heads *types.Fo
 
 	return &types.ForkchoiceUpdatedResult{PayloadStatus: types.PayloadStatusV1{Status: types.ExecutionValid, LatestValidHash: &heads.HeadBlockHash}, PayloadID: &id}, nil
 }
+
```

Comment:
Super minor suggestion but I think the code will be more maintainable longer-term if the `EthBackend` was moved to `eth.go` and the new `RPCMarshal` functions moved to `types/rpc_encoding.go`